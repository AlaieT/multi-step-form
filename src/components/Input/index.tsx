import React from "react";

import classNames from "../../utils/classNames";

import type { InputProps } from "../../types";

import styles from "../../styles/components/input.module.scss";

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({
    label, error, className, id, ...rest
  }, ref) => (
    <div className={classNames(styles.input, className)}>
      <div className={styles.badges}>
        <label htmlFor={id}>{label}</label>
        {error && <label htmlFor={id}>{error}</label>}
      </div>
      <input {...rest} id={id} ref={ref} className={error && styles.error} />
    </div>
  )
);

Input.displayName = "Input";

export default Input;
