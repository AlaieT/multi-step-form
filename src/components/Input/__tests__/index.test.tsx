import React from "react";
import { render, cleanup } from "@testing-library/react";

import Input from "..";

afterEach(cleanup);

describe("Input testing", () => {
  it("Snapshot tesing", () => {
    const { asFragment } = render(
      <Input id="name" label="Name" error="Name is required!" />
    );

    expect(asFragment).toMatchSnapshot();
  });
});
