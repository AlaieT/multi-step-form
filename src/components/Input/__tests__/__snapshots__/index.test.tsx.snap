// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input testing Snapshot tesing 1`] = `[Function]`;
